File: test/fixtures/csharp/post_form.cs
using System.Net.Http;

HttpClient client = new HttpClient();

HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "http://localhost:28139/post-to-me.php");

MultipartFormDataContent content = new MultipartFormDataContent();
content.Add(new StringContent("davidwalsh"), "username");
content.Add(new StringContent("something"), "password");
request.Content = content;

HttpResponseMessage response = await client.SendAsync(request);
response.EnsureSuccessStatusCode();
string responseBody = await response.Content.ReadAsStringAsync();

File: test/fixtures/php-guzzle/post_form.php
<?php
require 'vendor/autoload.php';

use GuzzleHttp\Client;

$client = new Client();

$response = $client->post('http://localhost:28139/post-to-me.php', [
    'multipart' => [
        [
            'name' => 'username',
            'contents' => 'davidwalsh'
        ],
        [
            'name' => 'password',
            'contents' => 'something'
        ]
    ]
]);
